### PR TITLE
more content, improved typography

### PR DESCRIPTION
**Pull request by narimiran**
Here are some screenshots (not the latest version but should be 99% the same, so it gives an idea about the changes):

### CODE REVIEW

File: jekyll/index.html [hunk 1]
```diff
@@ -9,22 +9,55 @@
       <div class="pure-u-1 pure-u-md-3-5 main-block">
         <h1>Efficient and expressive programming.</h1>
         <h2>
-            Nim is a systems and applications programming language.
-            Statically typed and compiled, it provides unparalleled performance
-            in an elegant package.
+          Nim is a statically typed, compiled, garbage-collected systems programming language
```
Comment (Araq) [inline jekyll/index.html]: Remove the "garbage-collected" here.

File: jekyll/index.html [hunk 2]
```diff
@@ -9,22 +9,55 @@
...
+          with a design that focuses on efficiency, expressiveness,
+          and elegance, providing unparalleled
```
Comment (Araq) [inline jekyll/index.html]: Remove "providing unparalleled performance in an elegant package", it's braggy and doesn't add anything to what was said earlier.

**Comment by dom96** [comment]
Hrm. Not sure about this. The huge list of features is incredibly dense and makes the "Features" page redundant.

Why not put this list in the "Features" page?

Another small thing:

* The sponsor money/names should be centered, the layout looks misaligned with your changes. Please revert it to be centered, removing that green background is a good change.

File: jekyll/index.html [hunk 3]
```diff
@@ -7,24 +7,50 @@
...
         <ul>
-          <li>High-performance garbage-collected language</li>
-          <li>Compiles to C, C++ or JavaScript</li>
-          <li>Produces dependency-free binaries</li>
-          <li>Runs on Windows, macOS, Linux, and more</li>
+          <li>Native code generation, not dependent on a virtual machine:
+              Nim produces <b>small executables without dependencies</b> for easy redistribution.</li>
+          <li>The Nim Compiler runs on Windows, Linux, BSD and Mac OS X. Porting to other platforms is easy.</li>
+          <li>Compiles to C, C++ or JavaScript.</li>
+          <li>A fast deferred reference counting memory management that supports soft real-time systems (like games).</li>
```
Comment (Araq) [inline jekyll/index.html]: Typo here, remove the 'A'.

File: jekyll/index.html [hunk 4]
```diff
@@ -7,24 +7,50 @@
...
+          <li>Compile time evaluation of user-defined functions.</li>
+          <li>Whole program dead code elimination: Only used functions are included in the executable.</li>
+          <li>Value-based datatypes: For instance, objects and arrays are allocated on the stack.</li>
         </ul>
+        <h3>Expressive</h3>
+        <ul>
+          <li>The Nim compiler and all of the standard library are <b>implemented in Nim.</b></li>
+          <li>Powerful AST-based hygienic macro system.</li>
+          <li>Macros can use the imperative paradigm to construct parse trees.</li>
+          <li>Macros cannot change Nim's syntax because there is no need for it.
```
Comment (Araq) [inline jekyll/index.html]: This bullet point should be the first bullect point under 'Elegant'.

File: jekyll/index.html [hunk 5]
```diff
@@ -7,24 +7,50 @@
...
+              Nim's syntax is flexible enough.</li>
+          <li>User-definable operators: code with new operators is often easier to read
```
Comment (Araq) [inline jekyll/index.html]: This bullet point should be before the macro-related bullet points.

**Comment by Araq** [comment]
> Why not put this list in the "Features" page?

The problem is that index.html is the entry point, not the "features" page and so not as many people read it.

**Comment by narimiran** [comment]
> The sponsor money/names should be centered, the layout looks misaligned with your changes.

Yep, my mistake there. Fixed.

----

> Why not put this list in the "Features" page?

Because "nobody clicks links".

Have you seen a new version of https://ziglang.org/ ? People are full of praise for it, because it offers lots of useful information at one place, and it is the first thing you see when you come to their site. I haven't heard no complaints that it is "incredibly dense" and/or that it should be 'hidden' behind some link.

> makes the "Features" page redundant.

Yes, that's my opinion too, and for me the course of action is: lets see what to do with it, but the index.html should definitely provide more information than it currently does. IMO, either we remove it or we update it to contain more information and examples. (But this is a discussion for some future PR)

**Comment by dom96** [comment]
Alright. Let's see how it goes, we can look at analytics to see if more people are visiting the docs/downloads page after your changes ;)

**Comment by andrewrk** [comment]
> Because "nobody clicks links".
> 
> Have you seen a new version of https://ziglang.org/ ?

FWIW somebody clicked that link because it ended up in my referrer URL log :wink: 

In all sincerity: the new screenshots look great - the two-column approach with features on the left and code examples on the right is appealing. I'd be curious to find out how the analytics fare.

**Comment by Nindaleth** [comment]
Thanks for revamping the landing page!

After the ribbon removal, now there is a direct GitHub repo link just at the end of one page, and nowhere else. Is that intended, compared to website sources being linked in every footer?

**Comment by narimiran** [comment]
> the new screenshots look great

Thanks :)
There is more that could be done/improved, but I'm leaving that for some other time....

----

> After the ribbon removal, now there is a direct GitHub repo link just at the end of one page, and nowhere else. Is that intended, compared to website sources being linked in every footer?

Thanks for noticing — it is fixed in the latest commit.

**Pull request merged**